src/ECModel.jl

```diff
@@ -852,6 +853,187 @@ function split_financial_terms(ECModel::AbstractEC, profit_distribution=nothing)
     )
 end
 
+function split_yearly_financial_terms(ECModel::AbstractEC, user_set_financial=nothing, profit_distribution=nothing)
+    gen_data = ECModel.gen_data
+    
+    project_lifetime = field(gen_data, "project_lifetime")
+
+    get_value = ((dense_axis, element) -> (if (element in axes(dense_axis)[1]) dense_axis[element] else 0.0 end))
+
+    year_set = 1:project_lifetime
+   
+    if isnothing(user_set_financial)
+        user_set_financial = [EC_CODE; get_user_set(ECModel)]
+    end

```

Comment:
By looking at the code below, since we need to calculate the reward here, we need the user_set_financial to match the user set. in the business plan, we could instead select different users

Reply:
I've changed some stuff. Do you think that we could add an input to the function. I'll try to finalize the EC as a whole and then I'll try